lib/router.js

```diff
@@ -210,47 +213,24 @@ Router.prototype._loadModule = function(modDef, globals) {
     }
 };
 
-Router.prototype._loadModules = function(node, hyperModules, scope) {
+Router.prototype._loadModules = function(node, hyperModules, scope, parentSegment) {
     var self = this;
-    if (Array.isArray(hyperModules)) {
-        throw new Error('Old style module config detected! '
-                + 'New format expects an object, not an array. '
-                + 'Please update your config.');
-    } else if (typeof hyperModules !== 'object') {
-        return P.resolve();
-    }
-
-    return P.each(Object.keys(hyperModules), function(modulePath) {
-        var pathMods = hyperModules[modulePath];
-        if (!Array.isArray(pathMods)) {
-            throw new Error('Error in module definition for ' + modulePath + '!\n'
-                    + 'Expected an array, got ' + pathMods);
-        }
-        return P.each(pathMods, function(m) {
-            // Share modules
-            return self._loadModule(m, scope.globals)
-            .then(function(module) {
-                if (!module) {
-                    throw new Error('Null return when loading module ' + modulePath);
-                }
-                if (!module.spec) {
-                    throw new Error('Module ' + modulePath + ' did not export a spec.');
-                }
-
-                if (module.resources) {
-                    // Resources array is shared between nodes,
-                    // so need to modify the array, not create a new with concat
-                    module.resources.forEach(function(res) {
-                        node.value.resources.push(res);
-                    });
-                }
-                var childScope = scope.makeChild({
-                    operations: module.operations,
-                    globals: module.globals,
+    return P.each(hyperModules, function(moduleDefinition) {
```

Comment:
It might be worth validating that hyperModules is indeed an Array, and throw a human-readable error if it is not. This would be more helpful to a spec writer than a random `P.each` failure.